pkg/agent/agent.go

```diff
@@ -319,7 +319,12 @@ func getAgentResource() (*apiV1.ResourceInstance, error) {
 
 	response, err := agent.apicClient.ExecuteAPI(coreapi.GET, agentResourceURL, nil, nil)
 	if err != nil {
-		return nil, err
+		// if agentResourceType is already a standard generic agent resource
+		if agentResourceType == agentTypesMap[agent.cfg.AgentType] {
+			return nil, err
+		}
+		AgentResourceType = ""
```

Comment:
You maybe didn't do this, but to me code gets confusing if you have variables w/ the same name differing in only the leading upper/lower character. It makes it look like a typo.